compiler-core/src/diagnostic.rs

```diff
@@ -48,7 +49,13 @@ impl Diagnostic {
             None => self.write_title(buffer),
         };
 
-        writeln!(buffer, "{}", self.text).expect("write_project");
+        let mut text = self.text.clone();
```

Comment:
Let's write to the buffer rather than cloning this string

Reply:
Should I write to the buffer and then also directly write to the buffer the hint?